src/WebJobs.Extensions.DurableTask/DurableTaskExtension.cs

```diff
@@ -90,12 +90,14 @@ public DurableTaskExtension()
         /// <param name="nameResolver">The name resolver to use for looking up application settings.</param>
         /// <param name="orchestrationServiceFactory">The factory used to create orchestration service based on the configured storage provider.</param>
         /// <param name="durableHttpMessageHandlerFactory">The HTTP message handler that handles HTTP requests and HTTP responses.</param>
+        /// <param name="lifeCycleNotificationHelper">The lifecycle notification helper used for custom orchestration tracking.</param>
         public DurableTaskExtension(
             IOptions<DurableTaskOptions> options,
             ILoggerFactory loggerFactory,
             INameResolver nameResolver,
             IOrchestrationServiceFactory orchestrationServiceFactory,
-            IDurableHttpMessageHandlerFactory durableHttpMessageHandlerFactory = null)
+            IDurableHttpMessageHandlerFactory durableHttpMessageHandlerFactory = null,

```

Comment:
By agreeing with your opinion, if `EventGridNotificationHelper` itself is solved by Dependency Injection, I think that it can be solved very smartly.
(Event Grid-related settings can be consolidated in one place)

Reply:
As stated in the other method, I can see why you used your approach, and I think that the use of the default parameter is fine here.